### PR TITLE
Analogic for date component

### PR DESCRIPTION
**Pull request by Tryxns**
This PR is to resolves Fred's request in PR https://github.com/Avaiga/taipy/pull/1991
*analogic for date range will come in separated PR
cc: @FredLL-Avaiga 

Result:
Screenshot 2024-11-05 at 12-28-28 Date - Example

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/DateSelector.tsx [hunk 1]
```diff
@@ -109,18 +115,38 @@ const DateSelector = (props: DateSelectorProps) => {
...
+                                    value={value}
+                                    onChange={handleChange}
+                                    className={getSuffixedClassNames(className, "-picker")}
+                                    disabled={!active}
+                                    slotProps={textFieldProps}
+                                    label={props.label}
+                                    format={props.format}
+                                    sx={dateSx}
+                                    viewRenderers={{
+                                        hours: renderTimeViewClock,
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/DateSelector.tsx]: apply the ternary operator (analogic ? ...) at the viewRenderers property level 
`viewRenderers={analogic ? analogicRenderers : undefined}`

With analogicRenderers defined outside the component

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
Almost there
Can you apply this to `date_range` too ?

File: frontend/taipy-gui/src/components/Taipy/DateSelector.tsx (hunk 1 above)
Comment (Tryxns) [reply frontend/taipy-gui/src/components/Taipy/DateSelector.tsx]: Done

**Comment by Tryxns** [comment]
Screenshot 2024-10-21 at 22-11-29 Analogic for date component · Avaiga_taipy@0b52cf5
I Just took a quick glance in my current failed Overall Test Workflow:
https://github.com/Avaiga/taipy/actions/runs/11443080274/job/31835011106?pr=2115

Error in conftest.py <- not a part of my changed files

**Comment by FredLL-Avaiga** [comment]
please update your branch

**Comment by FredLL-Avaiga** [comment]
I've updated your branch

**Comment by Tryxns** [comment]
ty Fred!

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
some test failing
      115 |         await waitFor(() => {
      116 |             const alertMessage = screen.queryByText("Test Alert");
    > 117 |             expect(alertMessage).not.toBeInTheDocument();

**Comment by Tryxns** [comment]
@FredLL-Avaiga The error is in Alert.spec.tsx <= not a part of my changed files. Correct me if I'm wrong

**Comment by FredLL-Avaiga** [comment]
> @FredLL-Avaiga The error is in Alert.spec.tsx <= not a part of my changed files. Correct me if I'm wrong

You're right sorry
I've updated your branch

**Review by FredLL-Avaiga — APPROVED** [review]

**Comment by FredLL-Avaiga** [comment]
Can you merge this PR @jrobinAV ?

**Pull request merged**